apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1149,6 +1123,27 @@ namespace MWMechanics
             break;
         }
 
+        case ESM::MagicEffect::CurePoison:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Poison);
+            break;
+        case ESM::MagicEffect::CureParalyzation:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Paralyze);
+            break;
+        case ESM::MagicEffect::CureCommonDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCommonDisease();
+            break;
+        case ESM::MagicEffect::CureBlightDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeBlightDisease();
+            break;
+        case ESM::MagicEffect::CureCorprusDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCorprusDisease();
+            break;
+        case ESM::MagicEffect::Dispel:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeAll(magnitude);
```

Comment:
It might be interesting to see how vanilla handles a non-100% version. I've only ever used the 100% one in testing. I'll check that out later. I'm kind of busy at work so it may be a few days before I work on a fix for this but when I start I will assign the issue to myself on the bug tracker. In case you want to do it then please do likewise and we can avoid overlapping work. Speaking of which, I have what I think should be a fix in the works for the "aquatic creatures don't stop combat with the player" bug, so I'll go mark myself as working on that.